pytorch_lightning/core/datamodule.py

```diff
@@ -97,6 +97,8 @@ def __init__(
         self._has_teardown_test = False
         self._has_teardown_predict = False
 
+        self._log_hyperparams = True
```

Comment:
- should this be part of the hyperparams mixin? 
- why is it protected? are users expected to toggle this on or off?

Reply:
> should this be part of the hyperparams mixin?

Could make it work. Would need an additional check for `_log_hyperparams` in Trainer.
```python
if not hasattr(self, "_log_hyperparams"):
```

> why is it protected? are users expected to toggle this on or off?

Nope, should be expected to toggle with the `save_hyperparameters` api.

Reply:
yes this should go to the mixin class, otherwise the LightningModule has a dependency on the Mixin interface, but this should not be, or we can't call it a mixin.  

we can simply add the initialization of this attribute to the mixin, right?

Reply:
Updated!

Reply:
eggsellent!